src/main/java/org/sonatype/repository/chef/internal/proxy/ChefProxyFacetImpl.java

```diff
@@ -0,0 +1,240 @@
+import org.sonatype.nexus.repository.transaction.TransactionalStoreBlob;
+import org.sonatype.nexus.repository.transaction.TransactionalTouchBlob;
+import org.sonatype.nexus.repository.transaction.TransactionalTouchMetadata;
+import org.sonatype.nexus.repository.view.Content;
+import org.sonatype.nexus.repository.view.Context;
+import org.sonatype.nexus.repository.view.Parameters;
+import org.sonatype.nexus.repository.view.Payload;
+import org.sonatype.nexus.repository.view.matchers.token.TokenMatcher;
+import org.sonatype.nexus.transaction.UnitOfWork;
+import org.sonatype.repository.chef.internal.AssetKind;
+import org.sonatype.repository.chef.internal.metadata.ChefAttributes;
+import org.sonatype.repository.chef.internal.util.ChefAttributeParser;
+import org.sonatype.repository.chef.internal.util.ChefDataAccess;
+import org.sonatype.repository.chef.internal.util.ChefPathUtils;
+import org.sonatype.repository.chef.internal.util.CookBookApiAbsoluteUrlRemover;
+
+import com.google.common.base.Joiner;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+import static org.sonatype.nexus.repository.storage.AssetEntityAdapter.P_ASSET_KIND;
+import static org.sonatype.repository.chef.internal.AssetKind.COOKBOOK;
+
+/**
+ * Chef {@link ProxyFacet} implementation.
+ *
+ * @since 0.0.1
+ */
+@Named
+public class ChefProxyFacetImpl
+    extends ProxyFacetSupport
+{
+  private ChefDataAccess chefDataAccess;
+
+  private ChefPathUtils chefPathUtils;
+
+  private ChefAttributeParser chefAttributeParser;
+
+  private CookBookApiAbsoluteUrlRemover cookBookApiAbsoluteUrlRemover;
+
+  @Inject
+  public ChefProxyFacetImpl(final ChefDataAccess chefDataAccess,
+                            final ChefPathUtils chefPathUtils,
+                            final ChefAttributeParser chefAttributeParser,
+                            final CookBookApiAbsoluteUrlRemover cookBookApiAbsoluteUrlRemover) {
+    this.chefDataAccess = checkNotNull(chefDataAccess);
+    this.chefPathUtils = checkNotNull(chefPathUtils);
+    this.chefAttributeParser = checkNotNull(chefAttributeParser);
+    this.cookBookApiAbsoluteUrlRemover = checkNotNull(cookBookApiAbsoluteUrlRemover);
+  }
+
+  // HACK: Workaround for known CGLIB issue, forces an Import-Package for org.sonatype.nexus.repository.config
+  @Override
+  protected void doValidate(final Configuration configuration) throws Exception {
+    super.doValidate(configuration);
+  }
+
+  @Nullable
+  @Override
+  protected Content getCachedContent(final Context context) throws IOException {
+    AssetKind assetKind = context.getAttributes().require(AssetKind.class);
+    switch (assetKind) {
+      case COOKBOOK:
+        TokenMatcher.State matcherState = chefPathUtils.matcherState(context);
+        return getAsset(chefPathUtils.buildCookbookPath(matcherState));
+      case COOKBOOKS_LIST:
+      case COOKBOOK_DETAILS:
+      case COOKBOOK_DETAIL_VERSION:
+      case COOKBOOKS_SEARCH:
+        return null;
```

Comment:
Question: Why return null here? It looks like we are caching this data?

Reply:
We are not caching any of it...YET! I'd like to talk with you and see which we'd like to cache. Right now it's all pass through as most of these are dynamic endpoints.

Reply:
So I think that means that we wouldn't be able to use NXRM if the remote went offline? Kinda defeats the point of a repo manager so I'd vote that we cache, but yea lets talk it through.